docs/images.md

```diff
@@ -0,0 +1,193 @@
+# Image
+An `Image` is an `Inkings` augmented with an optional rectangular
+expanse of pixels that can be the target of GPU rasterization or the
+source for scanout to the `Display` These pixels are not inside the
+Image structure. Instead, the system (i.e. GPU) owns the actual
+pixels.
+
+It is possible that we must support referring to the underlying pixel
+data in two different processes. If so, we need a mechanism to pass
+image data handles between processes using an operating system
+appropriate mechanism such as `gralloc` buffers or `IOSurface`
+objects. We should however attempt to satisfy the needs of Gojiraw
+clients without providing such a facility as (at least in my opinion)
+inter-process GPU image transport has brought enormous complexity to
+Chrome.
+
+When overlays are available, a presentable `Image` will own no pixel
+data itself. Instead, it will contain (as it is an `Inkings`) one or
+more children `Image` objects that each have their own GPU-resident
+pixel data that will be used for scanout.
+
+```go
+const (
+	_ = iota
+	OverlayCandidate = iota
+)
+
+type Image interface {
+	Load(r Rectangle, data []byte, opts ...interface{}) <-channel error
+	Unload(r Rectangle, data []byte, opts ...interface{}) <-channel error
+	Release()
+	NewImage(r Rectangle, opts ...interface{}) (Image, error)
+}
+
+func Zero() Image
+```
+
+## LoadImage
+`Load` fills `Rectangle r` with pixel `data` as controlled by the
+additional configuration options. The routine returns immediately. The
+returned channel indicates when the uploading process has completed.
+
+Options specify stride, packing etc. as suggested by the arguments to
+`glTexSubImage2D`.
+
+- [ ] Am I providing a rich enough API surface for copy minimization
```

Comment:
I don't understand. A handle to what? The target is the `Image` handle. Though it's conceivable that I should have made the API synchronous or pass the conduit in. API design style that I waver on: should we build the conduits, pass them in or expect users to put asynchronous tasks in go routines?a

I'd expect a usage pattern like this:

```go
image, err := Zero().NewImage(r)
if err != nil {
  // am sad... give up...
  return
}

inprogressImages := make(map[Image]<-conduit error)
inprogressImages[image] = image.Load()

// ...

for 
  select {
  case err := <-c:
    if err != nil {
      // fill image with something appropriate.
    }
}
```

Hmm. This is sort of stupid and convoluted. I think that I should provide the conduit to the call here and everywhere else.